packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -723,23 +739,30 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
     final Offset focalPointScene = _transformationController.toScene(
       details.localFocalPoint,
     );
-    _gestureType ??= _getGestureType(
-      !widget.scaleEnabled ? 1.0 : details.scale,
-      !_rotateEnabled ? 0.0 : details.rotation,
-    );
+
     if (_gestureType == _GestureType.pan) {
-      _panAxis ??= _getPanAxis(_referenceFocalPoint, focalPointScene);
+      // When a gesture first starts, it sometimes has zero scale and rotation
+      // despite being a two-finger gesture. Here the gesture is allowed to be
+      // reinterpreted as its correct type after originally being marked as a
+      // pan.
+      final _GestureType currentGestureType = _getGestureType(details);
+      if (currentGestureType != _GestureType.pan) {
+        _gestureType = currentGestureType;
+      }
+    } else {
+      _gestureType ??= _getGestureType(details);
     }
-
     if (!_gestureIsSupported(_gestureType)) {
       return;
     }
 
+    if (_gestureType == _GestureType.pan) {
+      _panAxis ??= _getPanAxis(_referenceFocalPoint, focalPointScene);
+    }
+
     switch (_gestureType) {
       case _GestureType.scale:
-        if (_scaleStart == null) {
-          return;
-        }
+        assert(_scaleStart != null);
```

Comment:
The change from an if to an assert here and below was just something I noticed while I was working on this. It's not related to the fix.